rust/otel-arrow-rust/src/otap/filter.rs

```diff
@@ -0,0 +1,309 @@
+                .expect("boolean arrays should have equal length");
+        let log_record_filter =
+            arrow::compute::and(&include_log_record_filter, &exclude_log_record_filter)
+                .expect("boolean arrays should have equal length");
+        let log_attr_filter =
+            arrow::compute::and(&include_log_attr_filter, &exclude_log_attr_filter)
+                .expect("boolean arrays should have equal length");
+
+        let (resource_attr_filter, scope_attr_filter, log_record_filter, log_attr_filter) = self
+            .sync_up_filters(
+                &logs_payload,
+                resource_attr_filter,
+                log_record_filter,
+                log_attr_filter,
+            )?;
+
+        // get the record batches we are going to filter
+        let resource_attrs = logs_payload
+            .get(ArrowPayloadType::ResourceAttrs)
+            .context(error::LogRecordNotFoundSnafu)?;
+        let log_records = logs_payload
+            .get(ArrowPayloadType::Logs)
+            .context(error::LogRecordNotFoundSnafu)?;
+        let log_attrs = logs_payload
+            .get(ArrowPayloadType::LogAttrs)
+            .context(error::LogRecordNotFoundSnafu)?;
+        let scope_attrs = logs_payload
+            .get(ArrowPayloadType::ScopeAttrs)
+            .context(error::LogRecordNotFoundSnafu)?;
+
+        // apply filters to the logs
+        let filtered_resource_attrs =
+            arrow::compute::filter_record_batch(resource_attrs, &resource_attr_filter)
+                .expect("columns should have equal length");
+        let filtered_log_records =
+            arrow::compute::filter_record_batch(log_records, &log_record_filter)
+                .expect("columns should have equal length");
+        let filtered_log_attrs = arrow::compute::filter_record_batch(log_attrs, &log_attr_filter)
+            .expect("columns should have equal length");
+        let filtered_scope_attrs =
+            arrow::compute::filter_record_batch(scope_attrs, &scope_attr_filter)
+                .expect("columns should have equal length");
+
+        logs_payload.set(ArrowPayloadType::ResourceAttrs, filtered_resource_attrs);
+        logs_payload.set(ArrowPayloadType::Logs, filtered_log_records);
+        logs_payload.set(ArrowPayloadType::LogAttrs, filtered_log_attrs);
+        logs_payload.set(ArrowPayloadType::ScopeAttrs, filtered_scope_attrs);
+
+        Ok(logs_payload)
+    }
+
+    fn sync_up_filters(
+        &self,
+        logs_payload: &OtapArrowRecords,
+        mut resource_attr_filter: BooleanArray,
+        mut log_record_filter: BooleanArray,
+        mut log_attr_filter: BooleanArray,
+    ) -> Result<(BooleanArray, BooleanArray, BooleanArray, BooleanArray)> {
+        // get the record batches we are going to filter
+        let resource_attrs = logs_payload
+            .get(ArrowPayloadType::ResourceAttrs)
+            .context(error::LogRecordNotFoundSnafu)?;
```

Comment:
These attribute record batches are optional, so at some point we'll need to handle the case here where they don't exist